### PR TITLE
[BUGFIX] Fix contextual components handling invalid component paths

### PR DESCRIPTION
**Pull request by Serabe**
Contextual components show a message when passed null or unknown component
paths.

### CODE REVIEW

File: packages/ember-htmlbars/tests/helpers/closure_component_test.js [hunk 1]
```diff
@@ -309,4 +310,70 @@ if (isEnabled('ember-contextual-components')) {
...
+      Mandarin
+    );
+    registry.register(
+      'template:components/-mandarin',
+      compile(`ni hao`)
+    );
+    registry.register(
+      'template:components/-hindi',
+      compile(`Namaste`)
+    );
```
Comment (mixonic) [inline packages/ember-htmlbars/tests/helpers/closure_component_test.js]: Can the above lines of this test be removed? They seem to be un-used.

File: packages/ember-htmlbars/tests/helpers/closure_component_test.js [hunk 2]
```diff
@@ -309,4 +310,70 @@ if (isEnabled('ember-contextual-components')) {
...
+
+    let template = compile(`{{component (component lookupComponent)}}`);
+    component = Component.extend({ container, template }).create();
+
+    expectAssertion(() => {
+      runAppend(component);
+    });
+  });
+
+  QUnit.test('raises an assertion when component path is null', function() {
```
Comment (mixonic) [inline packages/ember-htmlbars/tests/helpers/closure_component_test.js]: This test and the above are the same (even the same name), however this one asserts the message. Perhaps the previous test should be removed?

File: packages/ember-htmlbars/tests/helpers/closure_component_test.js [hunk 3]
```diff
@@ -309,4 +310,70 @@ if (isEnabled('ember-contextual-components')) {
...
+    template = compile(`{{component (component compName)}}`);
+    component = Component.extend({
+      container,
+      template,
+      compName: 'not-a-component'
+    }).create();
+
+    expectAssertion(() => {
+      runAppend(component);
+    }, `The component helper cannot be used without a valid component name. You used "not-a-component" via (component compName)`);
```
Comment (mixonic) [inline packages/ember-htmlbars/tests/helpers/closure_component_test.js]: :+1: awesome!!

File: packages/ember-htmlbars/tests/helpers/closure_component_test.js (hunk 2 above)
Comment (Serabe) [inline packages/ember-htmlbars/tests/helpers/closure_component_test.js]: Fixed

File: packages/ember-htmlbars/tests/helpers/closure_component_test.js (hunk 1 above)
Comment (Serabe) [inline packages/ember-htmlbars/tests/helpers/closure_component_test.js]: Fixed

**Comment by rwjblue** [comment]
https://github.com/emberjs/ember.js/pull/12440 fixed the node-land tests, merging...

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**